# Chile Censo 2017 - Data analysis 

<p><img src="http://www.censo2017.cl/wp-content/uploads/2016/01/logo-2.png"></p>
## Trabajo MDS-2oTrim. ==> Inmigracion en Chile

**Profesora:** Donoso I.<br>
**Alumnos:** Briceno, H.; Rovai, M.; Tessada G.

<br>

## The Dataset:
### Censo:
http://www.censo2017.cl
<br>

In [1]:
import pandas as pd
import numpy as np
import matplotlib
import matplotlib.pyplot as plt
%matplotlib inline

In [3]:
!ls ./Microdatos_Manzanas

Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Areas.csv
Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Categorias.csv
Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Comunas.csv
Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Geografia_Manzanas.csv
Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Provincias.csv
Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Regiones.csv
Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Urbano.csv
Censo2017_ManzanaEntidad_CSV\Censo2017_Manzanas.csv
Censo2017_ManzanaEntidad_CSV\LEAME.docx
Copy of Censo2017_Manzanas.xls


In [4]:
# Load dataset
data = './Microdatos_Manzanas/Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Areas.csv'
area = pd.read_csv(data, sep=';')
area.shape # (2, 2)

(2, 2)

In [5]:
area

AREA NOM_AREA
0     1   URBANO
1     2    RURAL

In [6]:
# Load dataset
data = './Microdatos_Manzanas/Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Categorias.csv'
categorias = pd.read_csv(data, sep=';')
categorias.shape # (14, 2)

(14, 2)

In [7]:
categorias

CAT_ENT              NOM_CAT_ENT
0         1                   Ciudad
1         2                   Pueblo
2         3                    Aldea
3         4                  Caserío
4         5      Asentamiento Minero
5         6    Asentamiento Pesquero
6         7  Fundo-Estancia-Hacienda
7         8          Parcela-Hijuela
8         9        Parcela de Agrado
9        10       Comunidad Indígena
10       12               Campamento
11       13   Veranada-Majada-Aguada
12       14                    Otros
13       15            Indeterminada

In [8]:
# Load dataset
data = "./Microdatos_Manzanas/Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Comunas.csv"
comunas = pd.read_csv(data, sep=';')
comunas.shape # (346, 2)

(346, 2)

In [9]:
comunas.head()

COMUNA     NOM_COMUNA
0    1101        IQUIQUE
1    1107  ALTO HOSPICIO
2    1401   POZO ALMONTE
3    1402         CAMIÑA
4    1403       COLCHANE

In [10]:
# Load dataset
data = './Microdatos_Manzanas/Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Provincias.csv'
provincias = pd.read_csv(data, sep=';')
provincias.shape # (54,2)

(54, 2)

In [11]:
provincias.head()

PROVINCIA NOM_PROVINCIA
0         11       IQUIQUE
1         14     TAMARUGAL
2         21   ANTOFAGASTA
3         22        EL LOA
4         23     TOCOPILLA

In [12]:
# Load dataset
data = './Microdatos_Manzanas/Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Regiones.csv'
regiones = pd.read_csv(data, sep=';')
regiones.shape # (15, 2)

(15, 2)

In [13]:
regiones

REGION                     NOM_REGION
0        1                       TARAPACÁ
1        2                    ANTOFAGASTA
2        3                        ATACAMA
3        4                       COQUIMBO
4        5                     VALPARAÍSO
5        6  LIBERTADOR BERNARDO O’HIGGINS
6        7                          MAULE
7        8                         BIOBÍO
8        9                      ARAUCANÍA
9       10                      LOS LAGOS
10      11                          AYSÉN
11      12                     MAGALLANES
12      13                  METROPOLITANA
13      14                       LOS RÍOS
14      15             ARICA Y PARINACOTA

In [14]:
# Load dataset
data = "./Microdatos_Manzanas/Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Urbano.csv"
urbano = pd.read_csv(data, sep=';')
urbano.shape # (4858, 8)

(4858, 8)

In [15]:
urbano.head()

REGION  PROVINCIA  COMUNA  DC  AREA  ZONA NOM_URBANO  CAT
0       1         11    1101   1     1     1    IQUIQUE    1
1       1         11    1101   1     1     2    IQUIQUE    1
2       1         11    1101   2     1     1    IQUIQUE    1
3       1         11    1101   2     1     2    IQUIQUE    1
4       1         11    1101   2     1     3    IQUIQUE    1

In [16]:
# Load dataset
data = './Microdatos_Manzanas/Censo2017_ManzanaEntidad_CSV\Censo2017_Identificación_Geográfica\Microdato_Censo2017-Geografia_Manzanas.csv'
geo = pd.read_csv(data, sep=';')
geo.shape # (180_499, 12)

(180499, 12)

In [17]:
geo.head()

REGION  PROVINCIA  COMUNA  AREA  DC  ZC_LOC NOM_LOCALIDAD  ID_ZONA_LOC  \
0      12        121   12101     1   1       1           NaN         9817   
1      12        121   12101     1   1       1           NaN         9817   
2      12        121   12101     1   1       1           NaN         9817   
3      12        121   12101     1   1       1           NaN         9817   
4      12        121   12101     1   1       1           NaN         9817   

   MZ_ENT NOM_ENTIDAD  CAT_ENT      ID_MANZENT  
0       1         NaN      NaN  12101011001001  
1       3         NaN      NaN  12101011001003  
2       6         NaN      NaN  12101011001006  
3       7         NaN      NaN  12101011001007  
4       8         NaN      NaN  12101011001008

In [18]:
# Load dataset
data = './Microdatos_Manzanas/Censo2017_ManzanaEntidad_CSV\Censo2017_Manzanas.csv'
manzanas = pd.read_csv(data, sep=';')
manzanas.shape # 180499, 47)

(180499, 47)

In [19]:
# manzanas.to_excel('manzanas.xlsx')

In [20]:
manzanas.head()

REGION  PROVINCIA  COMUNA  DC  AREA  ZC_LOC  MZ_ENT  ID_ZONA_LOC  \
0       1         11    1101   1     1       1       1         7849   
1       1         11    1101   1     1       1      10         7849   
2       1         11    1101   1     1       1      11         7849   
3       1         11    1101   1     1       1      12         7849   
4       1         11    1101   1     1       1      13         7849   

      ID_MANZENT  PERSONAS  ...   P03B_7 P03C_1 P03C_2 P03C_3 P03C_4 P03C_5  \
0  1101011001001        15  ...        0      0      0      0      0      0   
1  1101011001010        70  ...        0     15      0      0      0      0   
2  1101011001011        36  ...        0     10      0      0      5      0   
3  1101011001012        65  ...        1     18      1      4      1      0   
4  1101011001013        39  ...        0      9      0      0      0      0   

  P05_1 P05_2  P05_3  P05_4  
0     0     0      0      0  
1    15     0      0      0  
2    15     0      0      0  
3    24     0      0      0  
4     9     0      0      0  

[5 rows x 47 columns]

In [21]:
manzanas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180499 entries, 0 to 180498
Data columns (total 47 columns):
REGION         180499 non-null int64
PROVINCIA      180499 non-null int64
COMUNA         180499 non-null int64
DC             180499 non-null int64
AREA           180499 non-null int64
ZC_LOC         180499 non-null int64
MZ_ENT         180499 non-null int64
ID_ZONA_LOC    180499 non-null int64
ID_MANZENT     180499 non-null int64
PERSONAS       180499 non-null int64
HOMBRES        180499 non-null object
MUJERES        180499 non-null object
EDAD_0A5       180499 non-null object
EDAD_6A14      180499 non-null object
EDAD_15A64     180499 non-null object
EDAD_65YMAS    180499 non-null object
INMIGRANTES    180499 non-null object
PUEBLO         180499 non-null object
VIV_PART       180499 non-null int64
VIV_COL        180499 non-null int64
MATACEP        180499 non-null int64
MATREC         180499 non-null int64
MATIRREC       180499 non-null int64
VPOMP          180499 non-null

In [22]:
manzanas.PERSONAS.describe()

count    180499.000000
mean         97.363437
std         149.647870
min           0.000000
25%          35.000000
50%          63.000000
75%         105.000000
max        9197.000000
Name: PERSONAS, dtype: float64

In [23]:
len(manzanas.HOMBRES[manzanas.HOMBRES == '*'])

3413

In [24]:
len(manzanas.MUJERES[manzanas.MUJERES == '*'])

3413

Hay 3413 Hombres y mujeres que no són identificados. Se puede considerar la proporción chilena: 1.043 Mujer x 1.0 hombre o dejar 50% (#personas/2)<br>
Las formulas abajo, calculan el numero de Hombres e mujeres desconocidos utilizadose de los 50%

In [25]:
# *********** NOTA: REACER LAS FORMULAS CON .LOC ***************
manzanas['HOMBRES'] = manzanas.apply(lambda row: round((row['PERSONAS'] / 2), 0) if row['HOMBRES'] in ('*') else row['HOMBRES'], axis=1)
manzanas['MUJERES'] = manzanas.apply(lambda row: round((row['PERSONAS'] - row['HOMBRES']), 0) if row['MUJERES'] in ('*') else row['MUJERES'], axis=1)

In [26]:
manzanas['HOMBRES'] = manzanas['HOMBRES'].astype(int)
manzanas['MUJERES'] = manzanas['MUJERES'].astype(int)

In [27]:
POP_CHILE = manzanas.PERSONAS.sum()

In [28]:
POP_HOMBRES_CHILE = manzanas.HOMBRES.sum()

In [29]:
POP_MUJERES_CHILE = manzanas.MUJERES.sum()

In [30]:
VIVENDAS_CHILE = manzanas.TOTAL_VIV.sum()

In [31]:
manzanas.INMIGRANTES.describe()

count     180499
unique       400
top            0
freq       95642
Name: INMIGRANTES, dtype: object

In [32]:
len(manzanas)

180499

In [33]:
len(manzanas.INMIGRANTES[manzanas.INMIGRANTES == '*'])

49220

In [34]:
len(manzanas.INMIGRANTES[manzanas.INMIGRANTES == '0'])

95642

In [35]:
len(manzanas.INMIGRANTES[manzanas.INMIGRANTES != '0'])

84857

Hay 49220 manzanas sin informacion si las personas son imigrantes. Acá se las consideró cero!

In [36]:
manzanas['INMIGRANTES'] = manzanas.apply(lambda row: 0 if row['INMIGRANTES'] in ('*') else row['INMIGRANTES'], axis=1)
manzanas['INMIGRANTES'] = manzanas['INMIGRANTES'].astype(int)

In [37]:
POP_INMIGRANTES_CHILE = manzanas.INMIGRANTES.sum()

In [38]:
print("Numero de personas en Chile:", POP_CHILE)
print("Numero de hombres en Chile:", POP_HOMBRES_CHILE)
print("Numero de mujeres en Chile:", POP_MUJERES_CHILE)
print("Numero de inmigrantes en Chile:", POP_INMIGRANTES_CHILE)
print("Numero de vivendas en Chile:", VIVENDAS_CHILE)
print("Numero medio de personas por vivenda en Chile:", round(POP_CHILE/VIVENDAS_CHILE, 2))
print("Porcentaje de inmigrantes en Chile {}%".format(round((POP_INMIGRANTES_CHILE/POP_CHILE)*100, 2)))

Numero de personas en Chile: 17574003
Numero de hombres en Chile: 8596972
Numero de mujeres en Chile: 8977031
Numero de inmigrantes en Chile: 664120
Numero de vivendas en Chile: 6499355
Numero medio de personas por vivenda en Chile: 2.7
Porcentaje de inmigrantes en Chile 3.78%


### Zoom on Region Metroplitana

In [39]:
df_rm = manzanas[manzanas['REGION'] == 13]

In [40]:
df_rm.shape

(51070, 47)

In [41]:
POP_RM = df_rm.PERSONAS.sum()
POP_HOMBRES_RM = df_rm.HOMBRES.sum()
POP_MUJERES_RM = df_rm.MUJERES.sum()
VIVENDAS_RM = df_rm.TOTAL_VIV.sum()
POP_INMIGRANTES_RM = df_rm.INMIGRANTES.sum()

print("Numero de personas en RM:", POP_RM)
print("Numero de hombres en RM:", POP_HOMBRES_RM)
print("Numero de mujeres en RM:", POP_MUJERES_RM)
print("Numero de inmigrantes en RM:", POP_INMIGRANTES_RM)
print("Numero de vivendas en RM:", VIVENDAS_RM)
print("Numero medio de personas por vivenda en RM:", round(POP_RM/VIVENDAS_RM, 2))
print("Porcentaje de inmigrantes en RM {}%".format(round((POP_INMIGRANTES_RM/POP_RM)*100, 2)))

Numero de personas en RM: 7112808
Numero de hombres en RM: 3462008
Numero de mujeres en RM: 3650800
Numero de inmigrantes en RM: 457110
Numero de vivendas en RM: 2378442
Numero medio de personas por vivenda en RM: 2.99
Porcentaje de inmigrantes en RM 6.43%


In [42]:
df_rm_im = df_rm[df_rm['INMIGRANTES'] > 0]
df_rm_im.shape

(19227, 47)

In [43]:
df_rm_im.to_excel('inmigrantes_metropolitan.xlsx')

In [44]:
inm_com = df_rm.groupby(by=['COMUNA']).INMIGRANTES.sum()

In [45]:
inm_com = pd.DataFrame(inm_com)

In [46]:
inm_com.head()

INMIGRANTES
COMUNA             
13101        110732
13102          3620
13103          6022
13104          9223
13105          3368

In [47]:
comunas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 346 entries, 0 to 345
Data columns (total 2 columns):
COMUNA        346 non-null int64
NOM_COMUNA    346 non-null object
dtypes: int64(1), object(1)
memory usage: 5.5+ KB


In [48]:
com_rm = comunas[(comunas.COMUNA >= 13000) & (comunas.COMUNA < 14000)]

In [49]:
com_rm.head()

COMUNA   NOM_COMUNA
278   13101     SANTIAGO
279   13102    CERRILLOS
280   13103  CERRO NAVIA
281   13104     CONCHALÍ
282   13105    EL BOSQUE

In [50]:
com_rm = com_rm.set_index('COMUNA')

In [51]:
com_rm.head()

NOM_COMUNA
COMUNA             
13101      SANTIAGO
13102     CERRILLOS
13103   CERRO NAVIA
13104      CONCHALÍ
13105     EL BOSQUE

In [52]:
inm_rm = pd.concat([com_rm, inm_com], axis = 1)

In [53]:
inm_rm.sort_values('INMIGRANTES', ascending=False, inplace=True)

In [54]:
def replaceCaracteres(text):
    text = text.replace('á', 'a')
    text = text.replace('é', 'e')
    text = text.replace('í', 'i')
    text = text.replace('ó', 'o')
    text = text.replace('ú', 'u')
    text = text.replace('ñ', 'n')
    text = text.replace('ã', 'a')
    return text

def limpiaTexto(text):
    text = text.lower()
    text = replaceCaracteres(text)
    return text 

In [55]:
inm_rm['NOM_COMUNA'] = inm_rm.NOM_COMUNA.apply(limpiaTexto)

In [56]:
inm_rm.head()

NOM_COMUNA  INMIGRANTES
COMUNA                               
13101           santiago       110732
13114         las condes        30308
13108      independencia        30023
13127           recoleta        23329
13106   estacion central        22479

In [57]:
inm_rm.to_excel('inmigrantes_rm_comuna.xlsx')